Sources/GRPC/GRPCStatus.swift

```diff
@@ -20,20 +20,81 @@ import NIOHTTP2
 
 /// Encapsulates the result of a gRPC call.
 public struct GRPCStatus: Error {
-  /// The status message of the RPC.
-  public var message: String?
+  /// Storage for message/cause. In the happy case ('ok') there will not be a message or cause
+  /// and this will reference a static storage containing nil values. Making it optional makes the
+  /// setters for message and cause a little messy.
+  private var storage: Storage = .makeStorage(message: nil, cause: nil)
```

Comment:
Can we remove the default value here? It risks being unconditionally called if we do anything interesting with inlinability here, and we don't use it on any code path anyway.